ext/gd/gd.c

```diff
@@ -4108,27 +4108,50 @@ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type,
 	switch (image_type) {
 #ifdef HAVE_GD_JPG
 		case PHP_GDIMG_TYPE_JPG:
+			if (quality < -1 || quality > 100) {
+				zend_argument_value_error(3, "must be at between -1 and 100");
+				ctx->gd_free(ctx);
+				RETURN_THROWS();
+			}
 			gdImageJpegCtx(im, ctx, (int) quality);
 			break;
 #endif
 #ifdef HAVE_GD_WEBP
 		case PHP_GDIMG_TYPE_WEBP:
-			if (quality == -1) {
+			if (quality < -1) {
+				zend_argument_value_error(3, "must be at least -1");
```

Comment:
The "greater than or equal to" style is more commonly used with "at least" having only a few hits in the codebase.